go/vt/vtgate/planbuilder/operators/aggregation_pushing.go

```diff
@@ -39,10 +39,16 @@ func tryPushingDownAggregator(ctx *plancontext.PlanningContext, aggregator *Aggr
 	case *ApplyJoin:
 		if ctx.DelegateAggregation {
 			output, applyResult, err = pushDownAggregationThroughJoin(ctx, aggregator, src)
+			if applyResult != rewrite.SameTree && aggregator.Original {
+				aggregator.aggregateTheAggregates()
+			}
 		}
 	case *Filter:
 		if ctx.DelegateAggregation {
 			output, applyResult, err = pushDownAggregationThroughFilter(ctx, aggregator, src)
+			if applyResult != rewrite.SameTree && aggregator.Original {
+				aggregator.aggregateTheAggregates()
+			}
```

Comment:
this change is not clear, why is this called without error check and cannot be moved after the switch?

Reply:
refactored. easier to understand now?